### PR TITLE
fix: improve binary sensor code - add more sensors

### PR DESCRIPTION
**Pull request by aretakisv**
This PR tries to improve a bit more Pull #422 

- Convert repeated code into function.
- Add more devices

### CODE REVIEW

File: lib/Gateway.js [hunk 1]
```diff
@@ -1680,65 +1702,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
...
+              discoveredObjectId = valueId.propertyName
               break
             default:
           }
+          // cfg not there?
+          cfg = cfg ? copy(cfg) : copy(hassCfg.binary_sensor)
+          // correct payload from true/false to numeric values
           cfg = setBinaryPayloadFromSensor(cfg, valueId, off)
+          // finally update object_id
+          cfg.object_id = discoveredObjectId
```
Comment (robertsLando) [inline lib/Gateway.js:1774]: @billiaz Shouldn't this be:

```suggestion
          cfg.object_id += discoveredObjectId
```

File: lib/Gateway.js [hunk 2]
```diff
@@ -1680,65 +1702,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
...
+                  )
                   break
                 default:
               }
-              cfg.object_id = valueId.propertyName
-              break
-            case 'Water Alarm':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.MOISTURE
+              discoveredObjectId = valueId.propertyName
```
Comment (robertsLando) [inline lib/Gateway.js:1765]: Why? By default it is set to `property`

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: lib/Gateway.js (hunk 1 above)
Comment (billiaz) [reply lib/Gateway.js:1774]: we replace the object ID from the default. to fit the one we need to use

Comment (billiaz) [reply lib/Gateway.js:1774]: i agree, let me make some changes, as this  'binary_sensor_' prefix misses everywhere

File: lib/Gateway.js (hunk 2 above)
Comment (billiaz) [reply lib/Gateway.js:1765]: yeap! right! :)

File: lib/Gateway.js [hunk 3]
```diff
@@ -662,6 +662,23 @@ function setBinaryPayloadFromSensor (cfg, valueId, offStateValue = 0) {
...
   return cfg
 }
+
+/**
+ *
+ * @param {import('../types').HassDevice} cfg Hass discovery Configuration
+ * @param {string} devClass Choosen device class
+ * @param {boolean} reversePayload reverse payload order
+ */
+function setBinarySensorPayload (devClass, reversePayload = false) {
```
Comment (robertsLando) [inline lib/Gateway.js:672]:
```suggestion

/**
 * Create a binary sensor configuration with a specific device class
 *
 * @param {string} devClass Choosen device class
 * @param {boolean} reversePayload reverse payload order
 * @returns {import('../types').HassDevice} cfg Hass discovery Configuration for the binary sensor
 */
function getBinarySensorConfig (devClass, reversePayload = false) {
```

Change all the other references according to this

File: lib/Gateway.js [hunk 4]
```diff
@@ -1680,65 +1702,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
...
-              break
-            case 'Water Alarm':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.MOISTURE
+              discoveredObjectId = valueId.propertyName
               break
             default:
           }
+          // cfg not there?
+          cfg = cfg ? copy(cfg) : copy(hassCfg.binary_sensor)
```
Comment (robertsLando) [inline lib/Gateway.js:1770]:
```suggestion
          cfg = cfg || copy(hassCfg.binary_sensor)
```

No need to copy it here, it's already copied in `getBinarySensorConfig` function

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: lib/Gateway.js [hunk 5]
```diff
@@ -1680,65 +1703,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
...
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.MOISTURE
+              discoveredObjectId = valueId.propertyName
               break
             default:
           }
+          // cfg not there?
+          cfg = cfg || copy(hassCfg.binary_sensor)
+          // correct payload from true/false to numeric values
           cfg = setBinaryPayloadFromSensor(cfg, valueId, off)
```
Comment (robertsLando) [inline lib/Gateway.js:1773]:
```suggestion
          setBinaryPayloadFromSensor(cfg, valueId, off)
```

No need to re-assign cfg here, you pass it as reference

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Review by robertsLando — APPROVED** [review]
LGTM

**Pull request merged**